In [1]:
from warnings import filterwarnings
filterwarnings('ignore')

In [6]:
import seaborn as sns
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt

%matplotlib inline

In [12]:
df = pd.read_csv(r"C:\Users\NEERAJ\Downloads\My Projects\Etlhive Project\hockey.csv")

## *Checking the basic info of the Dataframe*

In [13]:
df.head()

Team Name  Year  Wins  Losses  OT Losses  Win %  Goals For (GF)  \
0       Boston Bruins  1990    44      24        NaN  0.550             299   
1      Buffalo Sabres  1990    31      30        NaN  0.388             292   
2      Calgary Flames  1990    46      26        NaN  0.575             344   
3  Chicago Blackhawks  1990    49      23        NaN  0.613             284   
4   Detroit Red Wings  1990    34      38        NaN  0.425             273   

   Goals Against (GA)  + / -  
0                 264     35  
1                 278     14  
2                 263     81  
3                 211     73  
4                 298    -25

In [19]:
# Checking the shape of the Dataframe
df.shape

(582, 9)

In [20]:
# Checking the coulumn names
df.columns

Index(['Team Name', 'Year', 'Wins', 'Losses', 'OT Losses', 'Win %',
       'Goals For (GF)', 'Goals Against (GA)', '+ / -'],
      dtype='object')

In [21]:
# Chicking the information of the dataframe 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 582 entries, 0 to 581
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Team Name           582 non-null    object 
 1   Year                582 non-null    int64  
 2   Wins                582 non-null    int64  
 3   Losses              582 non-null    int64  
 4   OT Losses           358 non-null    float64
 5   Win %               582 non-null    float64
 6   Goals For (GF)      582 non-null    int64  
 7   Goals Against (GA)  582 non-null    int64  
 8   + / -               582 non-null    int64  
dtypes: float64(2), int64(6), object(1)
memory usage: 41.1+ KB


In [22]:
df.describe()

Year        Wins      Losses   OT Losses       Win %  \
count   582.000000  582.000000  582.000000  358.000000  582.000000   
mean   2000.907216   36.943299   32.353952    7.460894    0.458519   
std       6.330674    8.926081    8.411439    3.610191    0.102237   
min    1990.000000    9.000000   11.000000    0.000000    0.119000   
25%    1996.000000   31.000000   27.000000    5.000000    0.390000   
50%    2001.000000   38.000000   31.000000    7.000000    0.463000   
75%    2007.000000   43.000000   37.000000   10.000000    0.534000   
max    2011.000000   62.000000   71.000000   18.000000    0.756000   

       Goals For (GF)  Goals Against (GA)       + / -  
count      582.000000          582.000000  582.000000  
mean       234.063574          234.063574    0.000000  
std         40.548915           42.510718   45.277458  
min        115.000000          115.000000 -196.000000  
25%        211.000000          207.000000  -27.000000  
50%        231.000000          232.500000    4.000000  
75%        254.000000          258.750000   31.000000  
max        369.000000          414.000000  144.000000

## *Cleaning the Dataframe and (Remove duplicates, Drop Missing Values column etc.)*

In [24]:
## First we check the null values in the dataset
df.isnull().sum()

Team Name               0
Year                    0
Wins                    0
Losses                  0
OT Losses             224
Win %                   0
Goals For (GF)          0
Goals Against (GA)      0
+ / -                   0
dtype: int64

#### *We can see the there are 224 missing value in OT Losses* 

In [27]:
## Here we see the missing percentage of the dataset
df.isnull().sum() *100 / len(df)

Team Name              0.000000
Year                   0.000000
Wins                   0.000000
Losses                 0.000000
OT Losses             38.487973
Win %                  0.000000
Goals For (GF)         0.000000
Goals Against (GA)     0.000000
+ / -                  0.000000
dtype: float64

In [34]:
# Checking the Dupluicated value in the dataframe
df.duplicated().sum()

0

#### *No duplicated values found in the Dataframe*